lib/9pfs/9pfs_vnops.c

```diff
@@ -1015,12 +1015,27 @@ static int uk_9pfs_symlink(struct vnode *dvp, const char *op, const char *np)
 static int uk_9pfs_ioctl(struct vnode *dvp, struct vfscore_file *fp,
 			 unsigned long com, void *data)
 {
-	switch (com) {
-	case TIOCGWINSZ:
-		return ENOTTY;
-	default:
+	/**
+	 * HACK: In binary compatibility mode, Ruby tries to set O_ASYNC,
+	 * which Unikraft does not yet support. If the `ioctl` call returns
+	 * an error, Ruby stops working, even if it does not depend on the
+	 * O_ASYNC being properly set.
+	 *
+	 * Until proper support, just return 0 in case an `FIONBIO` ioctl
+	 * request is done, and ENOTSUP for all other cases.
+	 *
+	 * Setting `ioctl` to a nullop will not work, since it is used by
+	 * interpreted languages (e.g. python3) to check if it should start
+	 * the interpretor or just read a file.
+	 *
+	 * For every `ioctl` request related to a terminal, return ENOTTY.
+	 */
+	if (com == FIONBIO)
 		return 0;
-	}
+	else if (IOCTL_REQ_IS_TTY(com))
```

Comment:
since you `return` if the previous condition is true, `else` is not needed here.

Reply:
I also wonder if we should call the macro `IOCTL_CMD_ISTTY(cmd)`, what do you think?

Reply:
Sure, makes sense. Renamed.